sklearn/ensemble/forest.py

```diff
@@ -89,6 +87,29 @@ def _parallel_build_trees(tree, forest, X, y, sample_weight, tree_idx, n_trees,
         sample_counts = np.bincount(indices, minlength=n_samples)
         curr_sample_weight *= sample_counts
 
+        if class_weight == 'bootstrap':
+            expanded_class_weight = [curr_sample_weight]
+            for k in range(y.shape[1]):
+                y_full = y[:, k]
+                classes_full = np.unique(y_full)
+                y_boot = y_full[indices]
+                classes_boot = np.unique(y_boot)
+                # Get class weights for the bootstrap sample
+                weight_k = compute_class_weight('auto', classes_boot, y_boot)
+                # Expand class weights to cover all classes in original y
+                # (in case some were missing from the bootstrap sample)
+                weight_k = np.array([weight_k[np.where(classes_boot == c)][0]
+                                     if c in classes_boot
+                                     else 0.
+                                     for c in classes_full])
```

Comment:
A clever use of np.choose should enable removing this for loop and making the corresponding code much faster (hint: use mode='clip').